juriscraper/opinions/united_states/state/nh.py

```diff
@@ -1,127 +1,142 @@
-            current_year=date.today().year
-        )
         self.court_id = self.__module__
-        self.link_path = 'id("content")/div//ul//li//a[position()=1]'
-        self.link_text_regex = re.compile(
-            r"(\d{4}-\d+(?!.*\d{4}-\d+))(?:,|\.|\s?)\s+(.*)", re.M
-        )
-        self.docket_name_pairs = False
+        # Parameters are included in URL because this court's API only allows
+        # GET requests. Current architecture for scrapers only includes
+        # parameters in requests with the POST method
+        self.url = "https://www.courts.nh.gov/content/api/documents?sort=field_date_posted%7Cdesc&page=1&size=30&purpose=1331"
+        self.parameters = {
+            "sort": "field_date_posted|desc",
+            "page": "1",
+            # Page documents size isn't limited to website defaults: 5, 10 or 25
+            "size": "30",
+            # Opinions have a "documentPurposes" value of "1331|"
+            "purpose": "1331",
+        }
 
-    def _get_case_names(self):
-        self.docket_name_pairs = self._get_anchor_docket_name_pairs()
-        return [pair["name"] for pair in self.docket_name_pairs]
-
-    def _get_download_urls(self):
-        path = "id('content')/div//ul//li//a[position()=1]"
-        download_url = []
-        for element in self.html.xpath(path):
-            url = element.xpath("./@href")[0]
-            download_url.extend([url])
-        return download_url
-
-    def _get_case_dates(self):
-        dates = []
-        path = 'id("content")/div//strong'
-        sub_path = (
-            "./following-sibling::ul[1]//li|../following-sibling::ul[1]//li"
+    def _process_html(self) -> None:
+        """
+        Expected 'documentName' values:
+            - 2020-0544 and 2020-0554, In re G.B.
+            - 2020-0528, 2020-0546, 2020-0548, In re S.A. & a.
+            - 2019-0283 In re R.M.
+            - 2017-0336 & 2019-0071  State of New Hampshire v. Daniel Turcotte
+            - 2013-0779, Ichiban Japanese Steakhouse, Inc. v. Kymberly Rocheleau, and 2013-0780, Ichiban Japanese Steakhouse, Inc. v. Samantha Greaney
+            - 2013-0403, 2013-0445, and 2013-0593, In re Guardianship of Madelyn B.
+            - 2013-0392. State of New Hampshire v. Kevin Balch
+            - 010-290  Eric Lee Knight v. Cheryl Ann Maher
+            - 10-436, New Hampshire Health Care Association & a. v. Governor & a.
+            - 2006-329, 2006-334, I/M/O State of NH and Estate of Frank Crabtree, III; I/M/O Katherine Crabtree & a.
+            - LD-2016-0005, Petition of Sanjeev Lath & a.
+            - LD-06-009, BOSSE'S CASE
+        """
+        name_re = (
+            r"(?P<docket>(((LD-|)\d{2,4}-\d{3,},\s*|)(LD-|)\d{2,4}-\d{3,}(,\s*(&|and)|\s*(&|and)|,)\s*|)(LD-|)\d{2,4}-\d{3,})"
+            r"(?P<name>.*)"
+        )
+        name_sec_re = (
+            r"(?P<name_1>.*)"
+            r"(?P<docket_2>(\s*(&|and)|,)\s*(LD-|)\d{2,4}-\d{3,})"
+            r"(?P<name_2>.*)"
         )
```

Comment:
I see what you are doing but this is overly complicated.

Reply:
Moved to a function for simplification.